### PR TITLE
streamlink: 2.4.0 -> 3.0.1

### PR DESCRIPTION
**Pull request by DeeUnderscore**
###### Motivation for this change
<https://github.com/streamlink/streamlink/releases/tag/3.0.1>, <https://github.com/streamlink/streamlink/releases/tag/3.0.0>

Upstream [bumped lxml because that version gets Windows wheels](https://github.com/streamlink/streamlink/commit/6cad2faf1f58cd2d9f8c986814f2c10472c429b9), so we should be fine patching out the requirement. 

###### Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by marsam** [comment]
@GrahamcOfBorg eval

**Comment by ElXreno** [comment]
Result of `nixpkgs-review pr 146461` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>streamlink</li>
    <li>streamlink-twitch-gui-bin</li>
  </ul>
</details>

File: pkgs/applications/video/streamlink/default.nix [hunk 1]
```diff
@@ -22,26 +22,21 @@ python3Packages.buildPythonApplication rec {
...
     pycryptodome
+    pysocks
     requests
-    iso-639
-    iso3166
     websocket-client
-    isodate
-    lxml
   ]) ++ [
     rtmpdump
```
Comment (bastimeyer) [inline pkgs/applications/video/streamlink/default.nix:33]: The RTMPDump dependency has been removed in 3.0.0

**Comment by bastimeyer** [comment]
There's still an rtmpdump reference at line 3. Couldn't add a comment there earlier, as it's not part of the diff.

**Comment by DeeUnderscore** [comment]
Bah, you're right, thank you

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 146461` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>streamlink</li>
    <li>streamlink-twitch-gui-bin</li>
  </ul>
</details>

**Review by thiagokokada — APPROVED** [review]

**Comment by thiagokokada** [comment]
Tested and seems to work fine.

**Pull request merged**